File: python_model/control_inputs.py
import pyglet

class ControlInputs:
    def __init__(self):
        self.pitch = 0.0  # [-1, +1] fwd/aft
        self.roll = 0.0       # [-1, +1] left/right
        self.rudder = 0.0        # [-1, +1] left/right
        self.spoiler = 0.0       # [0, 1] retracted/extended
        
    def set_controls(self, longitudinal, lateral, rudder, spoiler):
        self.pitch = max(-1.0, min(1.0, longitudinal))
        self.roll = max(-1.0, min(1.0, lateral))
        self.rudder = max(-1.0, min(1.0, rudder))
        self.spoiler = max(0.0, min(1.0, spoiler))

       

 

File: P3DInstruments/P3DWinch/control_inputs.cpp
#pragma once

#include <algorithm>
#include "control_inputs.h"


void ControlInputs::set_controls(float elevator, float aileron, float rudder, float spoiler, float brake, float release) {
	this->elevator = std::max(-1.0f, std::min(1.0f, elevator));
	this->aileron = std::max(-1.0f, std::min(1.0f, aileron));
	this->rudder = std::max(-1.0f, std::min(1.0f, rudder));
	this->spoiler = std::max(0.0f, std::min(1.0f, spoiler));
	this->brake = std::max(0.0f, std::min(1.0f, brake));
	this->release = std::max(0.0f, std::min(1.0f, release));
}


